### PR TITLE
Send events on rack updates and rollbacks

### PR DESCRIPTION
**Pull request by awsmsrc**

### CODE REVIEW

File: provider/aws/system.go [hunk 1]
```diff
@@ -292,5 +296,99 @@ func (p *AWSProvider) SystemSave(system structs.System) error {
...
+		select {
+		case err := <-waitch:
+			if err == nil {
+				event.Status = "success"
+				p.EventSend(event, nil)
+				return
+			}
+
+			if err != nil && err.Error() == "exceeded 120 wait attempts" {
+				p.EventSend(event, fmt.Errorf("couldn't determine rack update status, timed out"))
```
Comment (MiguelMoll) [inline provider/aws/system.go]: nit: How about just saying `timed out` here?

**Comment by awsmsrc** [comment]
[Track Rack Update Errors](https://trello.com/c/8K5fjJxr/1941-track-rack-update-errors)

**Comment by awsmsrc** [comment]
[Track Rack Update Errors](https://trello.com/c/8K5fjJxr/1941-track-rack-update-errors)

**Comment by awsmsrc** [comment]
fixes #658

**Comment by ddollar** [comment]
Strongly disapprove of using this shortcut approach again. There is an incredibly easy and correct way to do this on AWS.

**Comment by MiguelMoll** [comment]
Perhaps we could follow the strategy taken here? https://github.com/convox/rack/pull/1987

First issue top of mind is message won't be as detailed as it is now without some kind of state management?  I could take another look at since it's still fresh in my head.

**Comment by MiguelMoll** [comment]
Going to close this PR for now as we plan on going in another direction.

**Pull request closed without merging**